build-logic/src/main/groovy/net/fabricmc/yarn/buildlogic/task/JavadocLintTask.groovy

```diff
@@ -0,0 +1,103 @@
+package net.fabricmc.yarn.buildlogic.task
+
+import cuchaz.enigma.ProgressListener
+import cuchaz.enigma.translation.mapping.EntryMapping
+import cuchaz.enigma.translation.mapping.serde.MappingFileNameFormat
+import cuchaz.enigma.translation.mapping.serde.MappingSaveParameters
+import cuchaz.enigma.translation.mapping.serde.enigma.EnigmaMappingsReader
+import cuchaz.enigma.translation.mapping.tree.EntryTree
+import cuchaz.enigma.translation.representation.entry.Entry
+import cuchaz.enigma.translation.representation.entry.LocalVariableEntry
+import cuchaz.enigma.translation.representation.entry.MethodEntry
+import groovy.transform.CompileStatic
+import org.gradle.api.DefaultTask
+import org.gradle.api.GradleException
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.tasks.InputDirectory
+import org.gradle.api.tasks.TaskAction
+
+@CompileStatic
+class JavadocLintTask extends DefaultTask {
+    private static final String PARAMETER_TAG_START = "@param "
+
+    @InputDirectory
+    DirectoryProperty mappingDirectory = project.objects.directoryProperty()
+
+    JavadocLintTask() {
+        // Only run when inputs have changed
+        outputs.upToDateWhen { true }
```

Comment:
No? I think this will make groovy always treat this task as up-to-date and cause it to not run? If you depend on input you should just remove this up to date spec

Reply:
No, I've tested that this works properly. The default up-to-date checking does not work because this task has no outputs, but the task still gets run if it hasn't been run with these inputs before.

Reply:
If you're curious, the up-to-date checks are done like this (from [the gradle user guide](https://docs.gradle.org/6.8.3/userguide/more_about_tasks.html#sec:task_outcomes):

![gradle up-to-date checks](https://user-images.githubusercontent.com/6596629/113770595-f8970700-972a-11eb-9f0e-b5d49f84921c.png)

Since there are no outputs, this fits into the second category. Note that up-to-date checks check both the outputs (that I configure manually) *and* the inputs.